### ISSUE TITLE
`P3A` being re-enabled after upgrading even though it was disabled

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#19324]
## Description 

`P3A` is being re-enabled whenever an upgrade occurs even though a user has disabled `P3A` via the on-boarding or under `Brave shields & privacy` via `Settings`.

## Steps to reproduce 

* install `1.31.90 Chromium: 95.0.4638.69` and disable `P3A` via on-boarding (installed using GPS)
* go into the Privacy/Security settings and ensure that `Allow privacy-preserving product analytics (P3A)` is disabled
* upgrade to `1.31.91 Chromium: 95.0.4638.69` and check the same setting (notice that  `Allow privacy-preserving product analytics (P3A)` is enabled)

## Actual result 

`P3A` is being enabled whenever a user upgrades.

## Expected result

`P3A` should stay disabled after upgrading. We should be respecting the users choice.

## Issue reproduces how often 

100% reproducible using the STR. @SergeyZhukovsky also reproduced/confirmed the issue.

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? `Yes`
- Can you reproduce this issue with the current Play Store Beta version? `Yes` 
- Can you reproduce this issue with the current Play Store Nightly version? `Yes`

## Device details
   - Install type (ARM, x86): `ARM`
   - Device type (Phone, Tablet, Phablet): `Samsung S10+` (Phone)
   - Android version: `Android 11`
   
## Brave version

Basically any version after https://github.com/brave/brave-core/pull/7765.

### Website problems only

- Does the issue resolve itself when disabling Brave Shields? `N/A`
- Does the issue resolve itself when disabling Brave Rewards? `N/A`
- Is the issue reproducible on the latest version of Chrome? `N/A`

### Additional information

CCing @aekeus @bsclifton @SergeyZhukovsky @diracdeltas @PrivacyMatters

**Comment by rillian** [comment]
@deeppandya Are you able to diagnose this? I'm not familiar with the android installer mechanism.

**Comment by bsclifton** [comment]
@rillian looks like this was unintentionally introduced with https://github.com/brave/brave-core/pull/7765

Problem is in the logic here:
https://github.com/brave/brave-core/blob/877a6dbf1346f0a0327b11355ed69d4d1515232e/android/java/org/chromium/chrome/browser/app/BraveActivity.java#L480-L481

This is always evaluating to true basically

**Comment by rillian** [comment]
@bsclifton I figured it was something like that, but I'm still having trouble following. I didn't find any documentation on what `p3a_enabled_for_existing_users` means. Are the shared prefs not preserved across the update?

**Comment by GeetaSarvadnya** [comment]
Verification passed on `Vivo X70 Pro version Android 11 ` running `Bravemonoarm.64_apk_1.32.11`


<details>
<summary>Case 1: Upgrade from GPS to 1.32.11</summary>

</details>

<details>
<summary>Case 2: </summary>

</details>
